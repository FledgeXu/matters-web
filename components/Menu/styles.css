.menu {
  color: var(--color-black);
  border-radius: 2px;
  background: var(--color-white);
}

.spacing-0 {
  padding: 0;
}
.spacing-xxxtight {
  padding: var(--spacing-xxx-tight) 0;
}
.spacing-xxtight {
  padding: var(--spacing-xx-tight) 0;
}

.shadow-light {
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.08);
}
.shadow-default {
  box-shadow: 0 4px 12px 0 rgba(0, 0, 0, 0.05), 0 1px 4px 0 rgba(0, 0, 0, 0.08);
}
.shadow-dark {
  box-shadow: 0 12px 24px 0 rgba(0, 0, 0, 0.12),
    0 4px 12px 0 rgba(0, 0, 0, 0.05), 0 1;
}
