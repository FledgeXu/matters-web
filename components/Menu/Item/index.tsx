import classNames from 'classnames'
import React from 'react'

import styles from './styles.css'

type MenuItemSpacing = 'xtight' | 'tight' | 'default'

interface MenuItemProps {
  spacing?: [MenuItemSpacing, MenuItemSpacing]
  hoverBgColor?: 'green' | 'grey'

  style?: React.CSSProperties
  className?: string
}

const Item: React.FC<MenuItemProps> = ({
  spacing = ['xtight', 'default'],
  hoverBgColor = 'grey',

  style,
  children
}) => {
  const itemClasses = classNames({
    item: true,
    [`spacing-vertical-${spacing[0]}`]: true,
    [`spacing-horizontal-${spacing[1]}`]: true,
    [`bg-hover-${hoverBgColor}`]: true
  })

  return (
    <li className={itemClasses} style={style} role="menuitem">
      {children}
      <style jsx>{styles}</style>
    </li>
  )
}

export default Item
