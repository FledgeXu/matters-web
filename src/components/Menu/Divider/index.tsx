import React from 'react'

import styles from './styles.css'

export default ({}) => {
  return (
    <>
      <hr />
      <style jsx>{styles}</style>
    </>
  )
}
