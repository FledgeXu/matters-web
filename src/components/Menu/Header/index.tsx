import React from 'react'

import styles from './styles.css'

interface MenuItemProps {
  title: string | React.ReactNode
}

const Item: React.FC<MenuItemProps> = ({ title, children }) => {
  return (
    <li>
      <header>
        <span className="title">{title}</span>
        {children}
      </header>

      <style jsx>{styles}</style>
    </li>
  )
}

export default Item
