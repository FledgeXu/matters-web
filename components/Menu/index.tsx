import classNames from 'classnames'
import React from 'react'

import Divider from './Divider'
import Header from './Header'
import Item from './Item'
import styles from './styles.css'

interface MenuProps {
  width?: '168px' | '100%'
  spacing?: '0' | 'xxxtight' | 'xxtight'

  style?: React.CSSProperties
  className?: string
}

export class Menu extends React.PureComponent<MenuProps> {
  public static Item = Item
  public static Divider = Divider
  public static Header = Header

  public render() {
    const {
      width = '168px',
      spacing = 'xxxtight',
      className,
      style,
      children
    } = this.props

    const menuClasses = classNames({
      menu: true,
      [`spacing-${spacing}`]: true,
      [className || '']: !!className
    })

    return (
      <>
        <ul className={menuClasses} style={{ width, ...style }} role="menu">
          {children}
        </ul>
        <style jsx>{styles}</style>
      </>
    )
  }
}
