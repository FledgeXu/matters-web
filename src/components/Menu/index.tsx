import classNames from 'classnames'
import React from 'react'

import Divider from './Divider'
import Header from './Header'
import Item from './Item'
import styles from './styles.css'

interface MenuProps {
  width?: 'sm' | 'full' | 'md'
  spacing?: '0' | 'xxxtight' | 'xxtight'

  style?: React.CSSProperties
  className?: string
}

export class Menu extends React.PureComponent<MenuProps> {
  public static Item = Item
  public static Divider = Divider
  public static Header = Header

  public render() {
    const {
      width = 'sm',
      spacing = 'xxtight',
      className,
      style,
      children
    } = this.props

    const menuClasses = classNames({
      menu: true,
      [`width-${width}`]: true,
      [`spacing-${spacing}`]: true,
      [className || '']: !!className
    })

    return (
      <ul className={menuClasses} style={style} role="menu">
        {children}

        <style jsx>{styles}</style>
      </ul>
    )
  }
}
