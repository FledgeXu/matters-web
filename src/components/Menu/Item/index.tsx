import classNames from 'classnames'
import React from 'react'

import { Card, CardProps } from '~/components'

import styles from './styles.css'

type MenuItemProps = {} & CardProps

const MenuItem: React.FC<MenuItemProps> = ({
  children,
  spacing = ['tight', 'base'],
  bgHoverColor = 'grey-lighter',
  ...restProps
}) => {
  const menuItemClass = classNames({
    'menu-item': true
  })

  return (
    <li className={menuItemClass} role="menuitem">
      <Card spacing={spacing} bgHoverColor={bgHoverColor} {...restProps}>
        {children}
      </Card>

      <style jsx>{styles}</style>
    </li>
  )
}

export default MenuItem
