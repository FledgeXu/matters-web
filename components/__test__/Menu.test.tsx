import React from 'react'
import { render } from 'react-testing-library'

import { Menu } from '../Menu'

test('Menu default style match snapshot', () => {
  const { container } = render(<Menu />)
  expect(container.firstChild).toMatchSnapshot()
})
