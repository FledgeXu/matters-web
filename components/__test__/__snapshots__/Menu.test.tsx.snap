// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu default style match snapshot 1`] = `
<ul
  class="jsx-undefined menu spacing-xxxtight"
  role="menu"
  style="width: 168px;"
/>
`;
